src/SubdomainChecker.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+class SubdomainChecker
+{
+    private static $initiated = false;
+    private static $isSubdomain = false;
+    private static $originalDomain = '';
+    private static $subDomain = '';
+
+    public static function init()
+    {
+        if (!self::$initiated) {
+            self::init_hooks();
+        }
+
```

Comment:
Why did you choose to make this class static? Isn't your `init()` method basically a constructor?  Why not put all that logic and the `new CF\*` classes in a constructor? 

If you stick with this approach make sure to write tests for all these methods too.